File: easy/running_sum_1d/running_sum.py
class Solution(object):
    def runningSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        sums = []
        sums.append(nums[0])

        for i in range(1, len(nums)):
            sum = sums[i-1] + nums[i]
            sums.append(sum)

        return sums


s = Solution()
print("Solution 1 :", s.runningSum([1, 2, 3, 4]))
print("Solution 2 :", s.runningSum([1, 1, 1, 1, 1]))
print("Solution 3 :", s.runningSum([3, 1, 2, 10, 1]))

File: easy/array/running_sum_1d/running_sum.go
package main

import "fmt"

func runningSum(nums []int) []int {
	length := len(nums)
	sums := make([]int, 0, length)

	sums = append(sums, nums[0])
	for i := 1; i < length; i++ {
		sum := sums[i-1] + nums[i]
		sums = append(sums, sum)
	}

	return sums
}

func main() {
	fmt.Println("Solution 1 : ", runningSum([]int{1, 2, 3, 4}))
	fmt.Println("Solution 2 : ", runningSum([]int{1, 1, 1, 1, 1}))
	fmt.Println("Solution 3 : ", runningSum([]int{3, 1, 2, 10, 1}))
}
